README.md

```diff
@@ -64,6 +64,49 @@ However, processing all the datasets will take a long time and memory.
 
 *Note*: `poetry run` runs commands from within the virtual environment. You can also activate it with `poetry shell` and then simply run `etl ...`.
 
+#### Creating the pipeline of a new dataset
+
+These are the steps to create a data pipeline for a dataset called `example_dataset`, from an institution called
+`example_institution`, with version `YYYY-MM-DD` (where this date tag can typically be the current date when the dataset
+is being added to `etl`, or the date when the source data was released or updated):
+
+0. Activate the virtual environment (running `poetry shell`).
+1. Create a new branch in the `walden` submodule.
+2. If the source of data is a single file, you can manually download it and upload it as a new data snapshot into the
+S3 `walden` bucket.
+Alternatively, you can create an ingest script to do that (e.g. `etl/vendor/walden/ingests/example_institution.py`).
+The latter is preferable for example when the data is retrieved using an API.
+For more complex pipelines (e.g.
+[the FAOSTAT data pipeline](https://github.com/owid/etl/blob/master/docs/data/faostat.md)), the ingest script can
+download multiple files.
+Keep in mind that, if there is additional metadata, it should also be ingested into `walden` as part of the snapshot.
+3. Create an index file `etl/vendor/walden/index/example_institution/YYYY-MM-DD/example_dataset.json` for the new
```

Comment:
Manually creating an index file and manually uploading to S3 can be tricky and prone to errors. I think it's always better to write an ingest script & metadata yaml file which can be run from command line (`ggdc_maddison.py` is a good example). If for some reason we are not downloading the file from URL then ingest script should accept a `PATH` argument with path to a file we want to upload.

Reply:
I agree that it's much better to have a script. But to my knowledge, it should be possible to add things manually. This is important, for example, for datasets that are not accessible anywhere, like some piece of research data that appears in a paper but is not easily accessible from the web, or that was 'kindly provided by X in a private communication'. So, are you suggesting that I should not mention the manual upload, or that I should say that it's much preferable to use a script?

Reply:
You can mention manual upload, but I'd say it's better to use a script (especially because it's unclear where to upload those files on S3... to my knowledge no one has ever tried manual upload yet). So even if you have a file on your local I'd call
```
from owid.walden import add_to_catalog
add_to_catalog(metadata, "pathtofile", upload=True)
```

Reply:
OK, I've rephrased things a bit, and made that step a bit clearer.